platform/ios/MGLMapView.mm

```diff
@@ -2346,9 +2359,159 @@ - (void)removeOverlays:(NS_ARRAY_OF(id <MGLOverlay>) *)overlays
     [self removeAnnotations:overlays];
 }
 
-- (MGLAnnotationImage *)dequeueReusableAnnotationImageWithIdentifier:(NSString *)identifier
+- (nullable MGLAnnotationImage *)dequeueReusableAnnotationImageWithIdentifier:(NSString *)identifier
 {
-    return [self.annotationImages objectForKey:identifier];
+    // This prefix is used to avoid collisions with style-defined sprites in
+    // mbgl, but reusable identifiers are never prefixed.
+    if ([identifier hasPrefix:MGLAnnotationSpritePrefix])
+    {
+        identifier = [identifier substringFromIndex:MGLAnnotationSpritePrefix.length];
+    }
+    return self.annotationImagesByIdentifier[identifier];
+}
+
+/** Returns the straight-line distance between two coordinates. */
```

Comment:
This isn’t quite true. This function returns a value that correlates weakly to the distance between two coordinates. It’s only useful for sorting coordinates by proximity without going through the trouble of converting back and forth between screen and geographic coordinates. I’ll probably just inline it.